2048Solution/main/java/models/GameService.java

```diff
@@ -0,0 +1,230 @@
+package main.java.models;
+import java.util.ArrayList;
+import java.lang.Integer;
+public class GameService {
+    GameBoard board = new GameBoard();
+    GameUtility gameUtility = new GameUtility();
+    void initialize(int size){
+        board.setSize(size);
+        board.initializeBoard(size);
+        fillValueInBoard();
```

Comment:
Why called twice?

Reply:
because during initialisation for the first time we need two random positions.

Reply:
Got it. So abstract that logic by creating a function that calls this twice. Right now, it's difficult to understand the reason otherwise.